### PR TITLE
Add obsolete warnings to NServiceBus.Unity for last major version.

### PR DESCRIPTION
**Pull request by boblangley**
The [last Nuget package release of Unity](https://www.nuget.org/packages/Unity/) was 2015-10-16. Last commit to the [Unity repo](https://github.com/unitycontainer/unity) was 2016-03-18. Only one person has made commits to the project since it [was transferred from Microsoft to the community](https://blogs.msdn.microsoft.com/dotnet/2015/08/21/the-future-of-unity/). Any attempt to move it to .Net standard appears to be stalled. No community PRs have been accepted, and only one has even been commented on by the maintainers.

We should drop support for Unity and encourage our users to choose another container and migrate away from Unity. This PR would make it official.

### CODE REVIEW

File: src/NServiceBus.Unity.Tests/NServiceBus.ObjectBuilder.Unity.approved.cs [hunk 1]
```diff
@@ -2,17 +2,27 @@
 [assembly: System.Runtime.CompilerServices.InternalsVisibleToAttribute(@"NServiceBus.Unity.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100dde965e6172e019ac82c2639ffe494dd2e7dd16347c34762a05732b492e110f2e4e2e1b5ef2d85c848ccfb671ee20a47c8d1376276708dc30a90ff1121b647ba3b7259a6bc383b2034938ef0e275b58b920375ac605076178123693c6c4f1331661a62eba28c249386855637780e3ff5f23a6d854700eaa6803ef48907513b92")]
 [assembly: System.Runtime.InteropServices.ComVisibleAttribute(false)]
 [assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5.2", FrameworkDisplayName=".NET Framework 4.5.2")]
-
 namespace NServiceBus
 {
-    
+    [ObsoleteExAttribute(Message="NServiceBus.Unity has been deprecated. Using another container is advised.  Pleas" +
+        "e see https://docs.particular.net/nservicebus/containers/#supported-containers f" +
```
Comment (SimonCropp) [inline src/NServiceBus.Unity.Tests/NServiceBus.ObjectBuilder.Unity.approved.cs]: @boblangley note that by putting a link in the code results in the @Particular/container-maintainers ensuring that that link continues to exist, the content is still relevant to this obsolete, and the deep anchor continues exists. for the possible years that people may hit this obsolete

Comment (boblangley) [reply src/NServiceBus.Unity.Tests/NServiceBus.ObjectBuilder.Unity.approved.cs]: Yeah, I wasn't sure if it was a good idea. Do we have a protocol, or do we just avoid URLs all together?

Comment (SimonCropp) [reply src/NServiceBus.Unity.Tests/NServiceBus.ObjectBuilder.Unity.approved.cs]: generally avoid them. in other areas people have used text like "see the upgrade guide"

**Comment by boblangley** [comment]
Per the decision made in https://github.com/Particular/DotNetCoreLaunch/issues/34 I am withdrawing this PR.

**Pull request closed without merging**